cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/templates/etl/batch/ETLBatchTemplate.java

```diff
@@ -106,11 +117,78 @@ public void configureAdapter(String adapterName, ETLBatchConfig etlBatchConfig,
                                                    cronEntry));
   }
 
+  private void instantiateStages(ETLStage source, ETLStage sink, List<ETLStage> transformList)
+    throws InstantiationException, IllegalAccessException {
+    try {
+      String sourceClassName = sourceClassMap.get(source.getName());
+      String sinkClassName = sinkClassMap.get(sink.getName());
+      batchSource = (BatchSource) Class.forName(sourceClassName).newInstance();
+      batchSink = (BatchSink) Class.forName(sinkClassName).newInstance();
+
+      for (ETLStage etlStage : transformList) {
+        String transformName = transformClassMap.get(etlStage.getName());
+        Transform transformObj = (Transform) Class.forName(transformName).newInstance();
+        transforms.add(transformObj);
+      }
+    } catch (ClassNotFoundException e) {
```

Comment:
I think we can just catch any Exception here to catch everything. For example, an InstantiationException can happen if they give an interface, or give a class without nullary constructor, both of which would be invalid input.